Create Checklist and RadioItems labels with titles

giladmaya opened the pull request:
This PR fixes plotly/dash#1847 and adds a title to the Checklist and RadioItems options
## Contributor Checklist

- [x] I have broken down my PR scope into the following TODO tasks
   - Propagate title to label 
   - Implement integration tests  
- [x] I have run the tests locally and they passed. (refer to testing section in [contributing](https://github.com/plotly/dash/blob/master/CONTRIBUTING.md))
- [x] I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR

T4rk1n commented on components/dash-core-components/tests/integration/dropdown/test_dropdown_radioitems_checklist_option_title_prop.py:
```diff
@@ -0,0 +1,92 @@
+from dash import Dash, Input, Output, dcc, html
+
+
+@pytest.mark.DCC793
```
Non blocking, think we can remove that mark since it refers to old dcc repo issue.

T4rk1n commented on components/dash-core-components/tests/integration/dropdown/test_dropdown_radioitems_checklist_option_title_prop.py:
```diff
@@ -0,0 +1,92 @@
+# -*- coding: UTF-8 -*-
```
The filename is a bit long and there is now more than one component tested, maybe move up one directory and rename to `test_title_props.py` ?

T4rk1n commented on components/dash-core-components/tests/integration/dropdown/test_dropdown_radioitems_checklist_option_title_prop.py:
```diff
@@ -0,0 +1,92 @@
+
+
+@pytest.mark.DCC793
+@pytest.mark.parametrize(
```
Non blocking, parametrize is slow to run because it start/stop both the server and the browser. I think the test could just add all the components to the layout with different ids and performs the assertions statically.
For further refactors the callback is not really needed for testing purpose and the props can be given directly in the layout with a static title.

T4rk1n reviewed (commented):
Looks good, just a few non blocking comments on the test refactor and missing a changelog entry.

giladmaya commented:
> Looks good, just a few non blocking comments on the test refactor and missing a changelog entry.

I've addressed all your comments and pushed the changes.
The tests fail as a result of an issue (unrelated to my suggested changes) with Google Chrome installation.

Can you please re-review? Thank you

T4rk1n reviewed (approved):
:woman_dancing: Looks good :tada: 

The failing test is unrelated, I'll fix it.

giladmaya commented:
> 💃 Looks good 🎉
> 
> The failing test is unrelated, I'll fix it.

@T4rk1n I've tried re-triggering the failed tests but it seems like there's some instability with them, which as far as I can tell isn't related to my code changes. Can you please have a look? Thank you

T4rk1n commented:
> > 💃 Looks good 🎉
> > The failing test is unrelated, I'll fix it.
> 
> @T4rk1n I've tried re-triggering the failed tests but it seems like there's some instability with them, which as far as I can tell isn't related to my code changes. Can you please have a look? Thank you

We're just hitting some bad luck with the flaky tests, no worry I'll make them pass and merge it.
